loader/linux/icd_linux.c

```diff
@@ -61,76 +62,74 @@ void khrIcdOsVendorsEnumerate(void)
         // attempt to load all files in the directory
         for (dirEntry = readdir(dir); dirEntry; dirEntry = readdir(dir) )
         {
-            switch(dirEntry->d_type)
+            struct stat statBuff;
+            stat(dirEntry->d_name, &statBuff);
+            if(S_ISREG(statBuff.st_mode) || S_ISLNK(statBuff.st_mode))
             {
-            case DT_UNKNOWN:
-            case DT_REG:
-            case DT_LNK:
+                const char* extension = ".icd";
+                FILE *fin = NULL;
+                char* fileName = NULL;
+                char* buffer = NULL;
+                long bufferSize = 0;
+
+                // make sure the file name ends in .icd
+                if (strlen(extension) > strlen(dirEntry->d_name) )
```

Comment:
Is it in your coding style to have a space before the last `)` of an `if`?
I can see some, but not everywhere, so I do not know.